python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -379,7 +381,7 @@ def _wrap_config_type(
 
 
 class PartialIOManager(PartialResource[IOManager], IOManagerDefinition):
-    def __init__(self, resource_cls: Type[StructuredConfigIOManagerBase], data: Dict[str, Any]):
+    def __init__(self, resource_cls: Type[ConfigurableIOManagerBase], data: Dict[str, Any]):
```

Comment:
per discussion let's not say "ConfigurableIOManagerBase"

Proposals

`ConfigurableIOManagerFactory`
`ConfigurableIOManagerInjector`

I lean towards Injector since it is actually more specific than Factory. The Injector knows specially how to interact with our runtime. It "injects" a resource/IOManager into user-defined computations as runtime. `ConfigurableIOManagerFactory` sounds like it might create objects with type `ConfigurableIOManager` which is not true. It creates  objects with type`IOManager`.

Reply:
Cool. Given that recommend renaming `create_io_manager_to_pass_to_user_code` to `create_io_manager_to_inject_into_user_code`

Reply:
👍 (updated `create_object_to_pass_to_user_code` to `create_resource_to_inject_into_user_code` as well)